R/onevax_xvwrh.R

```diff
@@ -0,0 +1,153 @@
+       vax_t = c(0, t_stop),
+       vax_y = c(1, 0)
+  )
+}
+
+##' @name run_onevax_xvwrh
+##' @title run model with single vaccine for input parameter sets, either from
+##' initialisation or from equilibrium, those with waned vaccines are eligible
+##' for revaccination (R), and return to the R stratum
+##' @param vea_revax scalar or numeric vector with same length as `gono_params`
+##'  giving efficacy of revaccination against acquisition, default to same as
+##'  primary
+##' @param vei_revax scalar or numeric vector with same length as `gono_params`
+##'  giving efficacy of revaccination against infectiousness, default to same as
+##'  primary
+##' @param ved_revax scalar or numeric vector with same length as `gono_params`
+##'  giving efficacy of revaccination against duration of infection, default to
+##'  same as primary
+##' @param ves_revax scalar or numeric vector with same length as `gono_params`
+##'  giving efficacy of revaccination against symptoms, default to same as
+##'  primary
+##' @param dur_revax scalar or numeric vector with same length as `gono_params`
+##'  giving duration of protection for revaccination, default to same as primary
+##' @param hes Proportion of individuals in the population who are vaccine
+##'  hesitant
+##' @inheritParams run_onevax_xvwv
+##' @return A list of transformed model outputs
+##' @export
+run_onevax_xvwrh <- function(tt, gono_params, init_params = NULL,
+                            dur = 1e3, vea = 0, vei = 0, ved = 0, ves = 0,
+                            dur_revax = dur,
+                            vea_revax = vea, vei_revax = vei,
+                            ved_revax = ved, ves_revax = ves,
+                            vbe = 0, uptake = 0, strategy = "VbE",
+                            t_stop = 99, hes = 0) {
+
+  stopifnot(all(lengths(list(uptake, vea, vei, ved, ves, dur,
+                             vea_revax, vei_revax, ved_revax, ves_revax,
+                             dur_revax)) %in%
+                  c(1, length(gono_params))))
+
+  vax_params <- Map(vax_params_xvwrh, uptake = uptake, dur = dur,
+                    vea = vea, vei = vei, ved = ved, ves = ves,
+                    dur_revax = dur_revax,
+                    vea_revax = vea_revax, vei_revax = vei_revax,
+                    ved_revax = ved_revax, ves_revax = ves_revax,
+                    MoreArgs = list(strategy = strategy,
+                                    t_stop = t_stop, vbe = vbe))
+
+  if (is.null(init_params)) {
+
+    pars <- lapply(gono_params, model_params)
+    init_params <- lapply(pars, initial_params_xvwrh, hes = hes)
+
+    ret <- Map(run, gono_params = gono_params, vax_params = vax_params,
+               init_params = init_params,
+               MoreArgs = list(tt = tt))
+
+  } else {
+    ret <- Map(run, gono_params = gono_params, init_params = init_params,
+               vax_params = vax_params,
+               MoreArgs = list(tt = tt))
+  }
```

Comment:
```suggestion
  if (is.null(init_params)) {

    pars <- lapply(gono_params, model_params)
    init_params <- lapply(pars, initial_params_xvwrh, hes = hes)
    }

    ret <- Map(run, gono_params = gono_params, vax_params = vax_params,
               init_params = init_params,
               MoreArgs = list(tt = tt))
```
Lines 140 and 145 are identical so you can bring out of the if statement